### PR TITLE
[LEVWEB-1378] History page

### PR DESCRIPTION
**Pull request by RobinKnipe**
A new report page that shows the Cumulative Usage chart.

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,11 +51,11 @@
     "rollup-plugin-node-resolve": "^4.2.4",
     "rollup-plugin-replace": "^2.2.0",
     "rollup-plugin-terser": "^5.3.0",
-    "styled-components": "^4.2.1",
+    "styled-components": "^5.0.1",
```
Comment (daniel-ac-martin) [inline package.json:54]: How sure are you this doesn't break something in govuk-react?

File: test/lev-report/__snapshots__/History.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,3816 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (daniel-ac-martin) [inline test/lev-report/__snapshots__/History.test.js.snap:1]: It looks like we previously decided not to commit these files.

Do we need something added to `.gitignore`?

File: test/visualise.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
-import {renderToString} from "react-dom/server";
+const fs = require('fs');
+const { renderToString } = require('react-dom/server');
```
Comment (daniel-ac-martin) [inline test/visualise.js:4]: Why the move to `require` away from `import`?

File: jest.config.js [hunk 4]
```diff
@@ -9,14 +9,19 @@ module.exports = {
...
 
   // The glob patterns Jest uses to detect test files
   testMatch: [
-    path.resolve("./test/**/?(*.)+(spec|test).[tj]s?(x)")
+    path.resolve('./test/**/?(*.)+(spec|test).[tj]s?(x)')
   ],
 
+  transform: {
+    '^.+\\.js$': 'babel-jest',
+    '^.+\\.svg$': '<rootDir>/test/svg-transformer.js'
```
Comment (daniel-ac-martin) [inline jest.config.js:24]: Why has this been added?

File: package.json [hunk 5]
```diff
@@ -32,10 +32,10 @@
     "@babel/preset-env": "^7.8.7",
     "@babel/preset-react": "^7.8.3",
     "@svgr/rollup": "^4.3.0",
-    "babel-plugin-styled-components": "^1.10.0",
     "enzyme": "^3.11.0",
     "govuk-react": "^0.7.1",
     "jest": "^25.1.0",
+    "jest-styled-components": "^7.0.2",
```
Comment (daniel-ac-martin) [inline package.json:38]: What is the reason for this change?

File: jest.config.js (hunk 4 above)
Comment (daniel-ac-martin) [reply jest.config.js:24]: Okay I think I understand a little about what is going on from the commit message.

But do we really have to carry the code for this ourselves? Can't we use some babel plugin for this? Isn't that what was going on before?

I also can't see here a `.svg` file is being imported.

File: package.json (hunk 1 above)
Comment (RobinKnipe) [reply package.json:54]: image
That's from the horse's mouth, but also I have tested the changes.

File: test/lev-report/__snapshots__/History.test.js.snap (hunk 2 above)
Comment (RobinKnipe) [reply test/lev-report/__snapshots__/History.test.js.snap:1]: All `.snap` files must be committed along with the code, otherwise the tests won't work (or they will present false failures at least)

File: jest.config.js (hunk 4 above)
Comment (RobinKnipe) [reply jest.config.js:24]: Yes it's not an obvious one this... because this file checks a full page, not just a component, we have the header et al included as part of the output, which means the gov logo. I had a look at a couple of solutions for this, and this one allows us some control over what ends up in the sanpshot - at the moment it's just the filename, but it could also be the full SVG if we thought that were useful.

File: package.json (hunk 5 above)
Comment (RobinKnipe) [reply package.json:38]: This adds style information to the test snapshot, so we can spot style regressions.

File: test/visualise.js (hunk 3 above)
Comment (RobinKnipe) [reply test/visualise.js:4]: I'm happy to revert this, but I've been wanting to discuss styled components with you, which was what prompted this change.

File: package.json (hunk 1 above)
Comment (daniel-ac-martin) [reply package.json:54]: Hmm okay. Weird versioning...

File: test/lev-report/__snapshots__/History.test.js.snap (hunk 2 above)
Comment (daniel-ac-martin) [reply test/lev-report/__snapshots__/History.test.js.snap:1]: They don't seem to be in the repo at present. (Unless I was looking in the wrong place.)

File: package.json (hunk 5 above)
Comment (daniel-ac-martin) [reply package.json:38]: What about the removal of the babel plugin?

File: package.json [hunk 6]
```diff
@@ -32,10 +32,10 @@
     "@babel/preset-env": "^7.8.7",
     "@babel/preset-react": "^7.8.3",
     "@svgr/rollup": "^4.3.0",
-    "babel-plugin-styled-components": "^1.10.0",
```
Comment (daniel-ac-martin) [inline package.json:35]: I think this needs to go back in.

File: test/visualise.js (hunk 3 above)
Comment (daniel-ac-martin) [reply test/visualise.js:4]: It's not a big deal but it would be nice to be consistent.

File: test/lev-report/__snapshots__/History.test.js.snap (hunk 2 above)
Comment (RobinKnipe) [reply test/lev-report/__snapshots__/History.test.js.snap:1]: I can see the snapshots that have already been created here: https://github.com/UKHomeOffice/lev-react-components/tree/master/test/lev-report/charts/__snapshots__
This history one won't be in there until this PR is merged though.

File: test/visualise.js (hunk 3 above)
Comment (RobinKnipe) [reply test/visualise.js]: Agreed. Fixed.

File: package.json (hunk 6 above)
Comment (RobinKnipe) [reply package.json:35]: I've removed the counterpart from `.bablerc` and everything still seems to build correctly, so I guess it was never needed... or has been superseded by something else!?

File: test/lev-report/__snapshots__/History.test.js.snap (hunk 2 above)
Comment (daniel-ac-martin) [reply test/lev-report/__snapshots__/History.test.js.snap:1]: Okay.

File: package.json (hunk 6 above)
Comment (daniel-ac-martin) [reply package.json:35]: If you read the description for what it does: https://github.com/styled-components/babel-plugin-styled-components

It sounds like something that we want.

Comment (RobinKnipe) [reply package.json:35]: Yes, fair enough. I'll revert this...

Comment (RobinKnipe) [reply package.json]: Fixed
